networkx/algorithms/approximation/steinertree.py

```diff
@@ -248,3 +249,179 @@ def steiner_tree(G, terminal_nodes, weight="weight", method=None):
+
+    Parameters
+    ----------
+    G : DiGraph
+        A directed graph.
+    root : node
+        The root node.
+    terminals : set
+        Set of terminal nodes to be connected.
+    min_terminals : int
+        Minimum number of terminals to connect.
+    cutoff : int, optional
+        Maximum search depth from the root.
+    weight : string, optional (default="weight")
+        Edge attribute to use as weight.
+
+    Returns
+    -------
+    H : DiGraph
+        A directed Steiner tree subgraph.
+
+    Raises
+    ------
+    NetworkXError
+        If ``G`` is not directed, if root is not in ``G``, if cutoff
+        is not a positive integer or None, if ``min_terminals`` is
+        not a positive integer, if ``min_terminals`` exceeds the
+        number of given terminals, if ``min_terminals`` is not
+        a positive integer or if any terminals are not in ``G``.
+    NetworkXUnfeasible
+        If no terminals are given, if no terminals are reachable from
+        the root within the cutoff, or if the resulting tree fails
+        to cover at least ``min_terminals`` terminals.
+
+    References
+    ----------
+    .. [1] Charikar, M., Chekuri, C., Cheung, T-Y., Dai, Z., Goel, A.,
+           Guha, S., & Li, M. (1999). Approximation algorithms for
+           directed Steiner problems. *Journal of Algorithms*, 33(1), 73–91.
+           https://doi.org/10.1006/jagm.1999.1042
+    """
+    if not G.is_directed():
+        raise nx.NetworkXError("directed_steiner_tree only works for directed graphs")
+
+    if root not in G:
+        raise nx.NetworkXError(f"Root {root} not in G")
+
+    terminals = set(terminals)
+    if not terminals:
+        raise nx.NetworkXUnfeasible("No terminals given")
+
+    if cutoff is None:
+        cutoff = float("inf")
+    elif cutoff <= 0:
+        raise nx.NetworkXError("cutoff must be a positive integer or None")
+
+    if min_terminals is None:
+        min_terminals = len(terminals)
+    elif min_terminals <= 0:
+        raise nx.NetworkXError("min_terminals must be a positive integer or None")
+    elif min_terminals > len(terminals):
+        raise nx.NetworkXError(
+            "min_terminals must be less than or equal to the number of terminals"
+        )
+
+    missing = terminals - set(G.nodes)
+    if missing:
+        raise nx.NetworkXError(f"Terminals {missing} not in G")
+
+    reachable_terminals = terminals & _reachable_within(G, root, cutoff)
+    if not reachable_terminals:
+        raise nx.NetworkXUnfeasible(
+            f"No terminals are reachable from root {root} within cutoff={cutoff}"
+        )
+
+    H = _directed_steiner_tree(
+        G, root, reachable_terminals, min_terminals, cutoff, weight
+    )
+    covered = set(H.nodes) & set(reachable_terminals)
+    if len(covered) < min_terminals:
+        raise nx.NetworkXUnfeasible(
+            f"Directed Steiner tree could not cover at least {min_terminals} terminals {list(reachable_terminals)}"
```

Comment:
Just trying to keep the line length down, and state what is being printed.
```suggestion
            "Directed Steiner tree could not cover at least "
            f"{min_terminals}.\nReachable terminals: {list(reachable_terminals)}"
```

Reply:
Fixed in 8119c05d, thanks!